src/lib/one-graph/cli-client.js

```diff
@@ -282,24 +283,68 @@ const updateGraphQLOperationsFileFromPersistedDoc = async (input) => {
   witnessedIncomingDocumentHashes.push(hash)
 }
 
-const handleCliSessionEvent = async ({ event, netlifyGraphConfig, netlifyToken, schema, siteId }) => {
+const handleCliSessionEvent = async ({ event, netlifyGraphConfig, netlifyToken, schema, sessionId, siteId }) => {
   const { __typename, payload } = await event
   switch (__typename) {
     case 'OneGraphNetlifyCliSessionTestEvent':
-      await handleCliSessionEvent({ netlifyToken, event: payload, netlifyGraphConfig, schema, siteId })
+      await handleCliSessionEvent({ netlifyToken, event: payload, netlifyGraphConfig, schema, sessionId, siteId })
       break
-    case 'OneGraphNetlifyCliSessionGenerateHandlerEvent':
-      if (!payload.operationId || !payload.operationId.id) {
-        warn(`No operation id found in payload, ${JSON.stringify(payload, null, 2)}`)
+    case 'OneGraphNetlifyCliSessionGenerateHandlerEvent': {
+      if (!payload.operationId && !payload.operation.id) {
+        warn(`No operation id found in payload,
+  ${JSON.stringify(payload, null, 2)}`)
         return
       }
-      generateHandlerByOperationId({
+      const files = generateHandlerByOperationId({
         netlifyGraphConfig,
         schema,
-        operationId: payload.operationId.id,
+        operationId: payload.operationId || payload.operation.id,
         handlerOptions: payload,
       })
+
+      if (!files) {
+        warn(`No files generated for operation id: ${payload.operationId || payload.operation.id}`)
+        return
+      }
+
+      const config = loadNetlifyGraphConfig()
+      for (const file of files) {
+        const fileWrittenEvent = {
+          __typename: 'OneGraphNetlifyCliSessionFileWrittenEvent',
+          cliSessionId: sessionId,
+          payload: {
+            editor: config.editor,
```

Comment:
I suggest we use `config.editor || process.env.EDITOR`

Reply:
I'd like to avoid too much magic for the first iteration, but a good idea for down the road if this turns out to test well with users.